src/utils/time-format.js

```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
  */
 function toSecondsFromHHMMSS(input: string): number {
   const parts = input.split(':');
-  let seconds = 0;
   if (parts.length === 2) {
-    if (parseInt(parts[0]) > 59 || parseInt(parts[1]) > 59) {
-      return 0;
-    }
-    seconds += parseInt(parts[0]) * 60;
-    seconds += parseInt(parts[1]);
+    return _toSecondsFromArray(parts, false);
+  }
+  if (parts.length === 3) {
+    return _toSecondsFromArray(parts, true);
   }
+
+  return 0;
+}
+
+/**
+ * Handling the time parts array
+ *
+ * @param {Array<string>} parts the array
+ * @param {boolean} hasHours whether the parts array contains hours or not
+ * @returns {number} number of seconds
+ * @private
+ */
+function _toSecondsFromArray(parts: Array<string>, hasHours: boolean): number {
+  const minutesIndex = hasHours ? 1 : 0;
```

Comment:
i think that you can shorten it to something like

function _toSecondsFromArray(parts: Array<string>): number {
  let seconds = 0;
  let multiplier = 1;

  for (let i = parts.length - 1; i >= 0; --i) {
    seconds += parts[i] * multiplier;
    multiplier = multiplier * 60;
  }

  return seconds;
}

Reply:
cool thanks